planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp

```diff
@@ -348,7 +348,7 @@ std::tuple<double, double> OptimizationBasedPlanner::calcInitialMotion(
           "engage speed (%.3f) until vehicle speed reaches engage_vel_thr (%.3f). stop_dist = %.3f",
           vehicle_speed, target_vel, engage_velocity_, engage_vel_thr, stop_dist.value());
         return std::make_tuple(initial_vel, initial_acc);
-      } else if (stop_dist) {
+      } else {
```

Comment:
the type of stop_dist is std::optional<double>.
I am wondering if this warning is correct?

Reply:
or should we use `stop_dist.has_value()` ?

Reply:
The original if branch is like this:
```
if ( ... && !stop_dist) {
  ...
} else if (stop_dist) {
  ...
}
```
if ` ... && !stop_dist` if false, then I think `stop_dist` is true.
Isn't is correct?

Reply:
ah, you are right :bow:

Reply:
> or should we use stop_dist.has_value() ?

Yes if you are not sure how `!stop_dist` is evaluated.